src/c_api/c_api.cc

```diff
@@ -46,6 +47,32 @@ XGB_DLL int XGBRegisterLogCallback(void (*callback)(const char*)) {
   API_END();
 }
 
+XGB_DLL int XGBSetGlobalConfig(const char** name, const char** value, size_t num_param) {
```

Comment:
Another option is use JSON.  We have `booster.load/save_config`, maybe we can align with it.

Reply:
Didn't think of that. Yes, that would reduce the boilerplace a bit.

Reply:
@trivialfis One issue is that R does not have a built-in JSON parser, so we'd need to add a new dependency on the R package to add global config to the R package. On the other hand, the current approach does not require a new dependency.

Reply:
Implementing `get_config` in R turns out to be a headache, since I have to write R extension code that returns a dictionary. For the ease of maintenance, I propose that we make `jsonlite` a required dependency of XGBoost, so that `get_config` can simply return a JSON string.